### PR TITLE
Bump version of Mountpoint to v1.0.1

### PR DESCRIPTION
**Pull request by sauraank**
## Description of change
Preparing to bump Mountpoint to version v1.0.1.

Relevant issues: 
N/A
## Does this change impact existing behavior?
No

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I agree to the terms of the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).

### CODE REVIEW

File: mountpoint-s3/CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-## Unreleased changes
+## v1.0.1 (August 31, 2023)
 
 * Added new metrics for object writes, IO sizes, file handles, and directory operations. The existing `fuse.bytes_read` metric has been renamed to `fuse.total_bytes` and is now keyed by operation (`read`/`write`).
```
Comment (jamesbornholt) [inline mountpoint-s3/CHANGELOG.md:3]: We should add these PRs to the changelog:
- https://github.com/awslabs/mountpoint-s3/pull/475
- https://github.com/awslabs/mountpoint-s3/pull/467
- https://github.com/awslabs/mountpoint-s3/pull/489

File: mountpoint-s3/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,12 @@
 ## v1.0.1 (August 31, 2023)
 
+### Breaking changes
+* Made `allow_other` and `allow_root` mutually exclusive.
+
+### Other changes
 * Added new metrics for object writes, IO sizes, file handles, and directory operations. The existing `fuse.bytes_read` metric has been renamed to `fuse.total_bytes` and is now keyed by operation (`read`/`write`).
+* Close input/output handles when running in background.
+* Allowed reading restored GFR/GDA objects.
```
Comment (jamesbornholt) [inline mountpoint-s3/CHANGELOG.md:9]: Let's try to write full sentences for each of these about what changed and why customers should care. You can go read the PRs for a bit of context on each one. For example:
```suggestion
* Mountpoint can now read objects in the S3 Glacier Flexible Retrieval and S3 Glacier Deep Archive storage classes if they have been restored. Mountpoint cannot issue restore requests, but if you issue a restore request separately, the restored objects will be readable.
```

File: mountpoint-s3/CHANGELOG.md [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 
 ### Breaking changes
-* Made `allow_other` and `allow_root` mutually exclusive.
+* Made `allow_other` and `allow_root` mutually exclusive CLI arguments.
 
 ### Other changes
 * Added new metrics for object writes, IO sizes, file handles, and directory operations. The existing `fuse.bytes_read` metric has been renamed to `fuse.total_bytes` and is now keyed by operation (`read`/`write`).
-* Close input/output handles when running in background.
-* Allowed reading restored GFR/GDA objects.
+* Closing the input/output handles when we run Mountpoint in background mode. This prevents us from seeing anything the process prints after they are closed.
```
Comment (jamesbornholt) [inline mountpoint-s3/CHANGELOG.md:8]:
```suggestion
* When running in background mode (the default), Mountpoint now correctly closes standard input and output once mounting succeeds. This should fix issues with scripts that try to fork Mountpoint as a background process, which may previously have hung. ([#489](https://github.com/awslabs/mountpoint-s3/pull/489))
```

File: mountpoint-s3/CHANGELOG.md [hunk 4]
```diff
@@ -1,12 +1,12 @@
 ## v1.0.1 (August 31, 2023)
 
 ### Breaking changes
-* Made `allow_other` and `allow_root` mutually exclusive.
+* Made `allow_other` and `allow_root` mutually exclusive CLI arguments.
```
Comment (jamesbornholt) [inline mountpoint-s3/CHANGELOG.md:4]:
```suggestion
* The permissions CLI flags `--allow-other` and `--allow-root` are now mutually exclusive. `--allow-other` implies `--allow-root`, and so should be used if you want the effect of both flags. ([#475](https://github.com/awslabs/mountpoint-s3/pull/475))
```

File: mountpoint-s3/CHANGELOG.md [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
 
 ### Other changes
 * Added new metrics for object writes, IO sizes, file handles, and directory operations. The existing `fuse.bytes_read` metric has been renamed to `fuse.total_bytes` and is now keyed by operation (`read`/`write`).
-* Close input/output handles when running in background.
-* Allowed reading restored GFR/GDA objects.
+* Closing the input/output handles when we run Mountpoint in background mode. This prevents us from seeing anything the process prints after they are closed.
+* Mountpoint can now read objects in the S3 Glacier Flexible Retrieval and S3 Glacier Deep Archive storage classes if they have been restored. Mountpoint cannot issue restore requests, but if you issue a restore request separately, the restored objects will be readable.
```
Comment (jamesbornholt) [inline mountpoint-s3/CHANGELOG.md:9]:
```suggestion
* Mountpoint can now read objects in the S3 Glacier Flexible Retrieval and S3 Glacier Deep Archive storage classes if they have been restored. Mountpoint cannot issue restore requests, but if you issue a restore request separately, the restored objects will be readable. ([#467](https://github.com/awslabs/mountpoint-s3/pull/467))
```

File: mountpoint-s3/CHANGELOG.md [hunk 6]
```diff
@@ -1,12 +1,12 @@
 ## v1.0.1 (August 31, 2023)
 
 ### Breaking changes
-* Made `allow_other` and `allow_root` mutually exclusive.
+* Made `allow_other` and `allow_root` mutually exclusive CLI arguments.
 
 ### Other changes
 * Added new metrics for object writes, IO sizes, file handles, and directory operations. The existing `fuse.bytes_read` metric has been renamed to `fuse.total_bytes` and is now keyed by operation (`read`/`write`).
```
Comment (jamesbornholt) [inline mountpoint-s3/CHANGELOG.md:7]:
```suggestion
* Added new metrics for object writes, IO sizes, file handles, and directory operations. The existing `fuse.bytes_read` metric has been renamed to `fuse.total_bytes` and is now keyed by operation (`read`/`write`). ([#461](https://github.com/awslabs/mountpoint-s3/pull/461))
```

**Review by monthonk — APPROVED** [review]

**Pull request merged**